src/connector/src/source/pulsar/source/reader.rs

```diff
@@ -82,7 +82,9 @@ impl SplitReader for PulsarSplitReader {
 
 /// Filter offset for pulsar messages
 #[derive(Debug, Clone)]
-pub struct PulsarFilterOffset {
+struct PulsarFilterOffset {
+    /// Ledger ID of the message
+    pub ledger_id: u64,
```

Comment:
PulsarFilterOffset is not persisted, so it is safe not wrapping with an `Option`.